compiler/damlc/daml-opts/daml-opts/DA/Daml/Options.hs

```diff
@@ -466,6 +466,10 @@ setupDamlGHC mbProjectRoot options@Options{..} = do
   defaultCppPath <- liftIO locateCppPath
   modifyDynFlags $ adjustDynFlags options versionHeader tmpDir defaultCppPath
 
+  -- NOTE(MA): This ensures that the name supply inside HscEnv's NameCache
+  -- always starts with the same value (#14936)
+  liftIO $ initUniqSupply 0 1
```

Comment:
hours of smashing things together and staring at the screen and it turned out to be a single line fix

Reply:
huh does this not cause any other issues? iirc we call this at least once per file so we now reuse the same uniq supply even between files in the same project. I suppose you can argue this works like one-shot mode but I think the problem is that it doesn’t. one-shot mode works differently from how we process individual files.

Reply:
🤷 the tests will tell. I think it's fine because `DamlGhcSession` is defined with `defineEarlyCutoff`